fabfile.py

```diff
@@ -0,0 +1,64 @@
+import os
+import sys
+from fabric.api import *
+from fabric.colors import yellow, red
+
+env.hosts = [os.getenv('HATCH_API_URL')]
+
+CODE_DIR = '/srv/hatch/api'
+VENV = 'source /srv/venv-devhatch/bin/activate'
+
+@task
+def deploy(repo_uri=None):
+    """
+    Deploy latest version of Hatch API from master
+    """
+    if repo_uri is None:
+        repo_uri = 'git@github.com:pietdaniel/hatch-back-end.git'
+
+    puts(yellow('Running tests'))
+    with settings(warn_only=True):
+        if local('nosetests').failed:
+            puts(red('Tests failed!', bold=True))
+            puts(red('Will not proceed', bold=True))
+            return
+
+    puts(yellow('Deploying hatch'))
+
+    git_hash = "git ls-remote -h {} master | awk '{{print $1;}}' | cut -c -7".format(repo_uri)
```

Comment:
I didn't want to be dependent on being inside the git repo to have the hash and this seemed better than writing the hash to a file.